srcts/src/shiny/index.ts

```diff
@@ -618,6 +636,20 @@ class ShinyClass {
     initialValues[".clientdata_url_port"] = window.location.port;
     initialValues[".clientdata_url_pathname"] = window.location.pathname;
 
+    // Send useragent
+    initialValues[".clientdata_user_agent"] = navigator.userAgent;
+
+    // Send whether the client is mobile
+    initialValues[".clientdata_is_mobile"] = /Mobi|Android/i.test(
+      navigator.userAgent
+    );
+
+    // Send whether the client is desktop
+    initialValues[".clientdata_is_desktop"] =
+      !/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
+        navigator.userAgent
+      );
```

Comment:
After looking into this, I regret suggesting that we attempt this :)

The definition of a mobile device doesn't seem very well-defined, especially if basing that off of `userAgent`.

I would be more comfortable basing it off of the `window.innerWidth` (since that's how we determine it for collapsing layouts), but if you don't have an immediate need for it, I'd say we leave it out of this PR

Reply:
Agreed, removed from this PR